src/libs/vehicle/vehicle-factory.ts

```diff
@@ -106,10 +122,26 @@ function createVehicleFromMessage(message: Uint8Array): void {
 
   const heartbeat = mavlink_message.message as Message.Heartbeat
   if (heartbeat.autopilot.type === MavAutopilot.MAV_AUTOPILOT_INVALID) return
-  if (heartbeat.autopilot.type !== MavAutopilot.MAV_AUTOPILOT_ARDUPILOTMEGA) {
-    console.warn(`Vehicle not supported: ${system_id}/${component_id}: ${heartbeat.autopilot.type}`)
+  switch (heartbeat.autopilot.type) {
+    case MavAutopilot.MAV_AUTOPILOT_ARDUPILOTMEGA:
+      createArduPilotVehicle(heartbeat, system_id, component_id)
```

Comment:
One question here: is there a reason why we are creating this dedicated `createArduPilotVehicle` method instead of using the same `VehicleFactory.createVehicle(Vehicle.Firmware.ArduPilot, ...)` as it's being used with the PX4?

Reply:
We need to parse heartbeat.mavtype.type, that makes the code quite big inside this case, is for organization. We can move PX4 to a similar approach in the next PRs.